ingestors/kafka/src/main/scala/hydra/kafka/util/MetadataUtils.scala

```diff
@@ -0,0 +1,39 @@
+package hydra.kafka.util
+
+import hydra.kafka.model.{DataClassification, ObsoleteDataClassification}
+import spray.json.{JsString, JsValue}
+
+object MetadataUtils {
+
+  def oldToNewDataClassification(dataClassification: String): String = ObsoleteDataClassification.withNameOption(dataClassification) match {
+    case Some(ObsoleteDataClassification.InternalUseOnly)        => DataClassification.InternalUse.entryName
+    case Some(ObsoleteDataClassification.ConfidentialPII)        => DataClassification.Confidential.entryName
+    case Some(ObsoleteDataClassification.RestrictedEmployeeData) => DataClassification.Restricted.entryName
+    case Some(ObsoleteDataClassification.RestrictedFinancial)    => DataClassification.Restricted.entryName
+    case _                                                       => dataClassification
+  }
+
+  private def newToOldDataClassification(dataClassification: String): Option[String] = DataClassification.withNameOption(dataClassification) match {
+    case Some(DataClassification.Public)       => Some(DataClassification.Public.entryName)
+    case Some(DataClassification.InternalUse)  => Some(ObsoleteDataClassification.InternalUseOnly.entryName)
+    case Some(DataClassification.Confidential) => Some(ObsoleteDataClassification.ConfidentialPII.entryName)
+    case _                                     => None
+  }
+
+  def oldToNewDataClassification(dataClassification: Option[JsValue]): Option[JsValue] = dataClassification map {
+    case JsString(dc) => JsString(oldToNewDataClassification(dc))
+    case other        => other
+  }
+
+  def deriveSubDataClassification(dataClassification: String): Option[String] =
+    if (ObsoleteDataClassification.values.exists(_.entryName == dataClassification)) {
+      Some(dataClassification)
+    } else {
+      newToOldDataClassification(dataClassification)
+    }
```

Comment:
Should we always convert new-DC names to old-names ?

Reply:
Yes, newDC must be converted to oldDC as we are assigning the value to SubDataClassification field. The field SubDataClassification itself was created to store the old state of DataClassification.

Now, when the user send in New value for DataClassification, say 'InternalUse'. The SubDataClassification should store 'InternalUseOnly' which is an oldDC value. Therefore, in case where new values of DC are sent, corresponding old DC value must be stored in SubDataClassification.

The user are allowed to send oldDC value in DataClassification as well. In cases, DataClassification will converted and store newDC and the same oldDC value will be store in SubDataClassification.